aten/src/ATen/native/cuda/DistanceKernel.cu

```diff
@@ -335,12 +335,16 @@ void cdist_backward_kernel_impl(Tensor& result, const Tensor& grad, const Tensor
   const int block_x = 64;
   const int block_y = 16;
   const int grid_x = (m + block_x * 8 - 1) / (block_x * 8);
-  const int grid_y = (dist.numel() + block_y - 1) / block_y;
 
-  const dim3 grid(grid_x, grid_y);
+  const int64_t count = dist.numel();
+  const int64_t grid_temp = (count + block_y - 1) / block_y;
+
+  const int grid_y = (grid_temp - 1) / 65535 + 1;
+  const int grid_z = (grid_temp + grid_y - 1) / grid_y + 1;
```

Comment:
You don't need grid_y here:
```
In [1]: def grid_f(n): 
   ...:     y=(n - 1) // 65535 + 1; 
   ...:     z=(n + y - 1)//y + 1; 
   ...:     return y,z 
   ...:                                                                                                                                                                                                          

In [2]: grid_f(100 )                                                                                                                                                                                                        
Out[2]: (1, 101) # clearly should be 1, 100

In [3]: grid_f(65535)                                                                                                                                                             
Out[3]: (1, 65536) #this would be invalid configuration argument

In [4]: grid_f(65537)                                                                                                                                                                                            
Out[4]: (2, 32770) # 32770*2 = 65540, whereas 65538 would be enough

In [5]: def grid_f(n): 
   ...:     y=(n - 1) // 65535 + 1; 
   ...:     z=(n - 1)//y + 1; 
   ...:     return y,z 
   ...:                                                                                                                                                                                                          

In [6]: grid_f(100)                                                                                                                                                                                              
Out[6]: (1, 100) # 100, as expected

In [7]: grid_f(65535)                                                                                                                                                                                            
Out[7]: (1, 65535) #no illegal arg

In [8]: grid_f(65537)                                                                                                                                                                                            
Out[8]: (2, 32769) 2*32769 is 65538, minimum extra blocks
```

Reply:
Emmm seems I didn't fix the boundary cases nicely🤦‍♂️
Thank you!